coml/vis_utils.py

```diff
@@ -1,12 +1,438 @@
+    "answer": json.dumps({"channel": "y", "order": "descending"})
+}, {
+    "question": "Show the amount for all the payments processed with Visa distributing the payment date with a bar chart, I want to order in ascending by the amount of payment please.",
+    "answer": json.dumps({"channel": "y", "order": "ascending"})
+}]
+
+def parse_answer(response: str) -> str:
+    match = re.search(r"```.*```", response)
+    if match is not None:
+        answer = match.group(0)[3:-3]
+    else:
+        # Give up. Return whole response.
+        warnings.warn("Unable to parse the nl query from response.")
+        answer = response
+    return answer     
+
+def chart_summary(info: dict):
+    summarization = ""
+    if 'chart' in info:
+        summarization += f'''This is a {info['chart']} chart. '''
+    if 'title' in info:
+        summarization += f'''It's title is {info['title']}. '''
+
+    encoding = info['encoding']
+    for channel in encoding.keys():
+        if 'title' in encoding[channel]:
+            summarization += f'''The channel {channel} represents {encoding[channel]['title']} whose ''' 
+        else:
+            summarization += f'''Channel {channel}\'s'''
+        if encoding[channel]['type'] == 'quantitative':
+            summarization += f'''domain is {encoding[channel]['scale']['domain']}'''
+        else:
+            if 'ticks' in encoding[channel]['scale']:
+                summarization += f'''ticks is {encoding[channel]['scale']['ticks']}'''
+            else:
+                summarization += f'''domain is {encoding[channel]['scale']['domain']}'''
+        summarization += f''' and data type is {encoding[channel]['type']}.'''
+
+    return summarization
+
+
+PRECISION = 0.0001
+def spot_check(datum: dict, request:str, chart_info:dict, previous_code:str, variable_descriptions:dict, agent):
+    prerequisites = f'I\'m working on "{request}". You don\'t need to draw the visualization, just complete the transformation of the data. '
+
+    # Analyze the channel
+    data = chart_info['data']
+    encoding = chart_info['encoding']
+    quantitative_channels = []
+    other_channels = []
+    for channel in chart_info['encoding'].keys():
+        if encoding[channel]['type'] == 'quantitative':
+            quantitative_channels.append(channel)
+        else:
+            other_channels.append(channel)
+
+    # Choose a better way to query the data
+    if len(quantitative_channels) == 1:
+        ### generate query, while only one quantitative channel
+        query = f'Then find the {encoding[quantitative_channels[0]]["title"]} when '
+        conditions = [f'the {encoding[channel]["title"]} is {datum[encoding[channel]["field"]]}' for channel in other_channels]
+        query += ' and '.join(conditions)
+        query += '.'
+        value = {
+            'vis': datum[encoding[quantitative_channels[0]]['field']]
+        }
+
+
+        ### generate code to verify value
```

Comment:
It would be better to have this part as a method in agent called `answer_question`, maybe.